src/screens/PinFlow/CurrentPinScreen.tsx

```diff
@@ -14,37 +25,87 @@ interface Props {
   appSettings: {
     isPinSet: boolean;
   };
+  setTimeCounter: (timestamp: number) => SetTimeCounterAction;
+  setFailedAttempts: (attempt: number) => SetFailedAttemptsAction;
+  timeCounter: TimeCounterState;
 }
 
 interface State {
   pin: string;
   error: string;
+  failedTimes: number;
+  isCount: boolean;
 }
 
-export class CurrentPinScreen extends PureComponent<Props, State> {
+class CurrentPinScreen extends PureComponent<Props, State> {
   state = {
     pin: '',
     error: '',
+    failedTimes: 0,
+    isCount: true,
+  };
+
+  handleFailedAttempt = (increasedFailedTimes: number) => {
+    const { attempt } = this.props.timeCounter;
+    const isFinalAttempt = increasedFailedTimes > 2;
+    const finalAttempt = 3;
+    const firstAttempt = 0;
+    let currentDate = dayjs();
+    let blockedTimeInMinutes = 1;
+
+    if (attempt === 0) {
+      currentDate = currentDate.add(1, 'minute');
+      blockedTimeInMinutes = 1;
+    } else if (attempt === 1) {
+      currentDate = currentDate.add(2, 'minute');
+      blockedTimeInMinutes = 2;
+    } else if (attempt > 1) {
+      currentDate = currentDate.add(10, 'minute');
+      blockedTimeInMinutes = 10;
+    }
+
+    if (isFinalAttempt) {
+      this.props.setTimeCounter(currentDate.unix());
+      this.props.setFailedAttempts(attempt + 1);
+      this.setState({ isCount: true });
+    }
+
+    return increasedFailedTimes !== firstAttempt && increasedFailedTimes !== finalAttempt
+      ? `\n${i18n.onboarding.failedTimesErrorInfo} ${blockedTimeInMinutes} ${i18n.onboarding.minutes}\n${i18n.onboarding.failedTimes} ${increasedFailedTimes}/${finalAttempt}`
+      : '';
   };
 
   updatePin = (pin: string) => {
+    const { setFailedAttempts } = this.props;
     this.setState({ pin }, async () => {
       if (this.state.pin.length === CONST.pinCodeLength) {
         const setPin = await SecureStorageService.getSecuredValue('pin');
         if (setPin === this.state.pin) {
+          setFailedAttempts(0);
           this.props.navigation.navigate(Route.CreatePin, {
             flowType: FlowType.newPin,
           });
         } else {
+          const increasedFailedTimes = this.state.failedTimes + 1;
```

Comment:
read this from redux and you don't need failedTimes locally

Reply:
> read this from redux and you don't need failedTimes locally
@KedziaPawel 
That is not stored in redux, as I've written before:
`failedTimes and attempts are two different thigs, failedTimes are included in one attempt. So you have attempts and attempt could have maximum three failedTimes, then the timer is opening, attempt is increasing and failedTimes are reseting.`

Reply:
but this is the the same number  `const increasedFailedTimes = this.props. timeCounter.attempt + 1;` and it should work